### PR TITLE
Fix HTML markup in article pages

### PR DESCRIPTION
**Pull request by jmau111**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

The markup of the button "follow" seems incorrect  according to the HTML validator. This PR replaces the old markup with a button tag.

The PR fixes several other HTML errors as described by the commits. 

## Related Tickets & Documents

https://github.com/forem/forem/issues/13289

### CODE REVIEW

**Comment by jmau111** [comment]
the button markup needs to fixed, and, to me, using `button_tag` is the proper way here https://github.com/forem/forem/pull/14130/commits/a7331e53760998e804ed5bd1f12fe737c644305a and here https://github.com/forem/forem/pull/14130/commits/3e171fe3c80fa8df6a373951cb3cca133358b830, but the e2e failed with that, and I'm not sure why. Need help on this one if you can. Thanks.

**Comment by juliannatetreault** [comment]
Thank you so much for working on this, @jmau111! 🎉 

It appears that the failing specs on Travis ( `spec/liquid_tags/organization_tag_spec.rb` and `spec/liquid_tags/tag_tag_spec.rb`) are related to your `button_tag` change -- would you mind fixing these, please? Let me know if you need any help in the meantime and thanks again for the fix!

**Comment by aitchiss** [comment]
Hmmm, the test failures are genuine because when used in the liquid tags the `button_tag` method is undefined, e.g:

screenshot of article preview page with an error message stating that the button_tag method can't be found for the user liquid tag

I don't know enough about rails to know why that is 😅  But I'll summon a backend dev for review and I'm sure they'll be able to help!

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -122,12 +122,9 @@ def sanitized_sidebar(text)
...
 
-    tag :button, # Yikes
-        class: "crayons-btn follow-action-button #{classes} whitespace-nowrap",
-        data: {
-          :info => { id: followable.id, className: followable.class.name, style: style }.to_json,
-          "follow-action-button" => true
-        }
+    info   = { id: followable.id, className: followable.class.name, style: style }.to_json
+    button_tag("Follow", type: "button", data: { info: info },
+                         class: "crayons-btn follow-action-button #{classes} whitespace-nowrap")
```
Comment (rhymes) [inline app/helpers/application_helper.rb:127]: This doesn't exist in the helper methods as @aitchiss mentioned. The previous code resulted in: `<button properties></button>`

I suggestion you use the `tag` method like this:

```ruby
info = { id: followable.id, className: followable.class.name, style: style }
classes = "crayons-btn follow-action-button whitespace-nowrap #{classes}"
tag.button(name: :button, type: :button, data: { info: info }, class: classes)
```

see https://api.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-tag

ps. `to_json` isn't needed, they're already encoded as JSON:

> Data attribute values are encoded to JSON, with the exception of strings, symbols and BigDecimals. 
see

Comment (jmau111) [reply app/helpers/application_helper.rb]: Thanks a lot @rhymes 👍🏻, that's the e2e test I did not understand, but with the screenshot by @aitchiss I understand it breaks the editor. I've added your suggestion and the editor is ok. The HTML validator seems happy it. It produces `<button></button>` without any text, but it seems  it's added with JavaScript anyway:

forem-follow-btn

**Review by rhymes — APPROVED** [review]

**Review by juliannatetreault — APPROVED** [review]
Thanks so much for working on this, @jmau111! 🎉

**Pull request merged**